openhtf/util/threads.py

```diff
@@ -95,7 +95,7 @@ def async_raise(self, exc_type):
     assert self.is_alive(), 'Only running threads have a thread identity'
     result = ctypes.pythonapi.PyThreadState_SetAsyncExc(
         ctypes.c_long(self.ident), ctypes.py_object(exc_type))
-    if result == 0:
+    if result == 0 and self.is_alive():
       raise ValueError('Thread ID was invalid.', self.ident)
     elif result != 1:
       # Something bad happened, call with a NULL exception to undo.
```

Comment:
From the `PyThreadState_SetAsyncExc ` documentation:

> Returns the number of thread states modified; this is normally one, but will be zero if the thread id isn’t found.

Not sure when this would happen. But `SystemError` doesn't seem correct. Maybe `RuntimeError`?

Reply:
Sure changed it to a RuntimeError.  

Also noticed that result=0 would go into the elif, so change to greater than.